### DISCUSSION TITLE
Flux calculation

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25672] [Q&A General]
Hi All,

I would like to calculate the flux around a boundary and a block.
And hopefully, from the calculated flux, I would like to calculate the quantity by multiplying the time step size and the area. I would like to do these calculations to confirm that the quantity is conserved.

Please let me know if there are MOOSE apps do these for me.
S. Thomas

**Comment by GiudGiud** [comment]
Hello

Which kind of flux? Diffusive? Advective?
Which method? Finite element CG/DG? Finite volume?

There's a bunch of postprocessors to compute various fluxes already in MOOSE yes. Picking the right one is not always easy though, you want to make sure the definition of the flux matches in the postprocessor and the kernels

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
I am going into Postprocessors and take a look at them.
I do appreciate your help, as always.
S. Thomas

**Reply by grmnptr** [reply]
You can try some of these:
https://mooseframework.inl.gov/source/postprocessors/VolumetricFlowRate.html
https://mooseframework.inl.gov/source/postprocessors/BoundaryFluxPostprocessor.html
https://mooseframework.inl.gov/source/postprocessors/SideDiffusiveFluxIntegral.html

If you haven't seen it, we have a complete list for the available postprocessors in MOOSE. You are free to read their documentation and see if they are applicable to your problem.
https://mooseframework.inl.gov/source/
